conans/model/build_info.py

```diff
@@ -209,15 +210,55 @@ def name(self):
     def name(self, value):
         self._name = value
 
+    # TODO: Deprecate for 2.0. Only cmake and pkg_config generators should access this.
+    #  Use get_property for 2.0
     def get_name(self, generator):
-        return self.names.get(generator, self._name)
-
+        property_name = None
+        if "cmake" in generator:
+            property_name = "cmake_target_name"
+        elif "pkg_config" in generator:
+            property_name = "pkg_config_name"
+        return self.get_property(property_name, generator) or self.names.get(generator, self._name)
+
+    # TODO: Deprecate for 2.0. Only cmake generators should access this. Use get_property for 2.0
     def get_filename(self, generator):
-        result = self.filenames.get(generator)
+        result = self.get_property("cmake_file_name", generator) or self.filenames.get(generator)
         if result:
             return result
         return self.get_name(generator)
 
+    # TODO: Deprecate for 2.0. Use get_property for 2.0
+    def get_build_modules(self):
```

Comment:
It is a bit undesired that this is called so many times, it could be cached, but the above is redefining ``self.build_modules = BuildModulesDict.from_list(self.get_build_modules())``, which is used in the method, so that might break such caching.

Recall that things in CppInfo might be called mamy many times while propagating graph, so this is one of the bottlenecks in graph resolution, we should be careful with adding here inefficient things.